spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/execution/ExtendedIcebergStrategy.scala

```diff
@@ -18,26 +18,26 @@
 
 package com.netease.arctic.spark.sql.execution
 
-import com.netease.arctic.spark.sql.catalyst.plans.ReplaceArcticData
+import com.netease.arctic.spark.sql.catalyst.plans.{AppendArcticData, OverwriteArcticData, OverwriteArcticDataByExpression, ReplaceArcticData}
 import com.netease.arctic.spark.table.ArcticSparkTable
 import com.netease.arctic.spark.{ArcticSparkCatalog, ArcticSparkSessionCatalog}
 import org.apache.iceberg.spark.{Spark3Util, SparkCatalog, SparkSessionCatalog}
-import org.apache.spark.sql.catalyst.analysis.{NamedRelation, ResolvedTable}
-import org.apache.spark.sql.catalyst.expressions.{And, Expression, NamedExpression}
+import org.apache.spark.sql.catalyst.analysis.NamedRelation
+import org.apache.spark.sql.catalyst.expressions.{And, Expression, NamedExpression, PredicateHelper}
 import org.apache.spark.sql.catalyst.planning.PhysicalOperation
+import org.apache.spark.sql.catalyst.plans.CreateArcticTableAsSelect
 import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.utils.TranslateUtils
 import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
 import org.apache.spark.sql.connector.iceberg.read.SupportsFileFilter
-import org.apache.spark.sql.execution.command.CreateTableLikeCommand
 import org.apache.spark.sql.execution.datasources.v2._
 import org.apache.spark.sql.execution.{FilterExec, LeafExecNode, ProjectExec, SparkPlan}
 import org.apache.spark.sql.util.CaseInsensitiveStringMap
 import org.apache.spark.sql.{SparkSession, Strategy}
 
-import scala.collection.JavaConverters
 import scala.collection.JavaConverters.{mapAsJavaMapConverter, seqAsJavaList}
 
-case class ExtendedIcebergStrategy(spark: SparkSession) extends Strategy {
+case class ExtendedIcebergStrategy(spark: SparkSession) extends Strategy with PredicateHelper{
```

Comment:
those code shoud be in ExtendedArcticStrategy. 

ExtendedIcebergStrategy is simply copy from iceberg code